pkg/service/issuing/model.go

```diff
@@ -0,0 +1,149 @@
+	Schema string `json:"schema"`
+
+	// Date that will be used to determine credential claims.
+	Data CredentialTemplateData `json:"data"`
+
+	// Parameter to determine the expiry of the credential.
+	Expiry TimeLike `json:"expiry"`
+
+	// Whether the credentials created should be revocable.
+	Revocable bool `json:"revocable"`
+}
+
+type IssuanceTemplate struct {
+	// ID of the credential manifest that this template corresponds to.
+	CredentialManifest string `json:"credentialManifest"`
+
+	// ID of the issuer that will be issuing the credentials.
+	Issuer string `json:"issuer"`
+
+	// Info required to create a credential from a credential application.
+	Credentials []CredentialTemplate `json:"credentials"`
+}
+
+type GetIssuanceTemplateResponse struct {
+	// The template that was requested.
+	IssuanceTemplate IssuanceTemplate `json:"issuanceTemplate"`
+}
+
+type CreateIssuanceTemplateRequest struct {
+	// ID to be used when creating the issuance template. Must be unique.
+	// Required.
+	ID string `json:"id" validate:"required"`
+
+	// The template to create.
+	IssuanceTemplate IssuanceTemplate `json:"issuanceTemplate"`
+}
+
+type DeleteIssuanceTemplateRequest struct {
+	// ID of the template that will be deleted.
+	// Required.
+	ID string `json:"id" validate:"required"`
+}
+
+type ListIssuanceTemplatesRequest struct {
+	// A parsed filter expression conforming to https://google.aip.dev/160.
+	Filter filtering.Filter
+}
+
+type ListIssuanceTemplatesResponse struct {
+	// The issuance templates that satisfy the query conditions.
+	IssuanceTemplates []IssuanceTemplate `json:"issuanceTemplates"`
+}
+
+type Service struct {
+	config  config.IssuingServiceConfig
+	storage Storage
+}
+
+func (s *Service) Type() framework.Type {
+	return framework.Issuing
+}
+
+func (s *Service) Status() framework.Status {
+	return framework.Status{Status: framework.StatusReady}
+}
+
+func NewIssuingService(config config.IssuingServiceConfig, s storage.ServiceStorage) (*Service, error) {
+	issuingStorage, err := NewIssuingStorage(s)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating issuing storage")
+	}
+	return &Service{
+		storage: *issuingStorage,
+		config:  config,
+	}, nil
+}
+
+type Storage struct {
+	db storage.ServiceStorage
+}
+
+func NewIssuingStorage(s storage.ServiceStorage) (*Storage, error) {
+	if s == nil {
+		return nil, errors.New("s cannot be nil")
+	}
+	return &Storage{
+		db: s,
+	}, nil
+}
+
+func (s *Service) GetIssuanceTemplate(request *GetIssuanceTemplateRequest) (*GetIssuanceTemplateResponse, error) {
+	return nil, nil
+}
+
+func (s *Service) CreateIssuanceTemplate(request *CreateIssuanceTemplateRequest) (*IssuanceTemplate, error) {
+	return nil, nil
+}
+
+func (s *Service) DeleteIssuanceTemplate(request *DeleteIssuanceTemplateRequest) error {
+	return nil
+}
+
+func (s *Service) ListIssuanceTemplates(request *ListIssuanceTemplatesRequest) (*ListIssuanceTemplatesResponse, error) {
```

Comment:
We have this as Get in other places, I do kind of like List better.

Eventually we can change everything to be List with a filter if it makes sense

Reply:
Will normalize in a future PR.